source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -228,14 +228,20 @@ std::vector<absl::string_view> Filter::getAlpnProtocols(const unsigned char* dat
     return protocols;
   }
 
-  absl::string_view str(reinterpret_cast<const char*>(data));
   for (unsigned int i = 0; i < len;) {
-    uint32_t protocol_length = 0;
-    protocol_length <<= 8;
-    protocol_length |= data[i];
-    ++i;
-    absl::string_view protocol(str.substr(i, protocol_length));
-    protocols.push_back(protocol);
+    uint8_t protocol_length = data[i++];
```

Comment:
So the format is that data is prefixed with a single byte, which represents the number of data bytes, correct? Can you add code-level comment to the method to document Filter::getAlpnProtocols while we are at it?

Tangentially related, but imho out of scope for now -- I'd love it if we could refactor this method into it's own unit of compilation under `/maistra/source/tls/alpn_util.cc` or some such.. and then we could have unit tests for this in `/maistra/source/tls/test_alpn_util.cc`.  That would prove correctness, tease out issues when run under asan, as well as serve as semi-documentation on assumptions and edge cases etc.

Reply:
Added comment to explain the format of the data being unmarshalled.

Reply:
👍 thank you, resolving.